### PR TITLE
feat(t04):search-job-by-title

### PR DESCRIPTION
**Pull request by VoChiNghia**
Link deploy :
http://tangy-baseball.surge.sh/

### CODE REVIEW

File: src/components/Header/Header.tsx [hunk 1]
```diff
@@ -52,8 +66,10 @@ const Header = () => {
                       type="search"
                       placeholder="Search"
                       aria-label="Search"
+                      value={searchInput}
+                      onChange={(e) => setSearchInput(e.target.value)}
                     />
-                    <button data-testid="button-search" className="btn btn-danger" type="submit">
+                    <button data-testid="button-search" className="btn btn-danger" type="button" onClick={(e) => handleSearch(e)} >
```
Comment (ManhNDM) [inline src/components/Header/Header.tsx:72]: Shlould be onClick={handleSearch}

File: src/components/Header/header.test.tsx [hunk 2]
```diff
@@ -1,6 +1,11 @@
 import React from "react";
 import { render,screen } from "@testing-library/react";
 import Header from "./Header";
+
+jest.mock("react-redux", () => ({
+  ...jest.requireActual("react-redux"),
+  useDispatch: jest.fn()
+}));
```
Comment (ManhNDM) [inline src/components/Header/header.test.tsx:8]: why use this?

File: src/components/ShowAllJobs/ShowAllJobs.tsx [hunk 3]
```diff
@@ -7,21 +7,32 @@ import { JobModal } from "../../type/type";
...
-  
+
   const { allJobs, loading } = useSelector(
     (state: RootState) => state.jobReducer
   );
+  const { searchValue } = useSelector((state: RootState) => state.searchJob);
+
+  let results = search(allJobs, searchValue);
+  results = allJobs
+
```
Comment (ManhNDM) [inline src/components/ShowAllJobs/ShowAllJobs.tsx:25]: Please handle in the reducer.

File: src/components/Header/Header.tsx [hunk 4]
```diff
@@ -1,5 +1,19 @@
...
+
 const Header = () => {
+  const [searchInput,setSearchInput] = useState<string>('')
+  const dispatch:DispatchType = useDispatch()
+
+  const handleSearch = (e:React.MouseEvent<HTMLElement>) => {
+    e.preventDefault()
+    dispatch(searchValueReducer(searchInput))
+    setSearchInput("")
+
```
[the remark this answers is not in the document]
Comment (VoChiNghia) [reply src/components/Header/Header.tsx]: i've fixed

File: src/components/Header/Header.tsx (hunk 1 above)
Comment (VoChiNghia) [reply src/components/Header/Header.tsx]: i've fixed

File: src/components/Header/Header.tsx [hunk 5]
```diff
@@ -52,8 +66,10 @@ const Header = () => {
                       type="search"
                       placeholder="Search"
                       aria-label="Search"
+                      value={searchInput}
+                      onChange={(e) => setSearchInput(e.target.value)}
```
[the remark this answers is not in the document]
Comment (VoChiNghia) [reply src/components/Header/Header.tsx]: i've fixed

File: src/components/Header/header.test.tsx (hunk 2 above)
Comment (VoChiNghia) [reply src/components/Header/header.test.tsx]: i've removed

**Pull request merged**